### DISCUSSION TITLE
Refreshing a table - DONE

### DISCUSSION THREAD

**Discussion by VilobatirKedis** [Budibase/budibase#2294] [Ideas]
Hello,

I have a table component that shows all the databases present in the MySQL server connected. I have also a button that creates a new database in the server. The problem is that the table doesn't update itself alone when I add new databases, so I need to update the page to see the results. How can I set up a button that updates the table for me?

I tried setting it up like this:

![Cattura](https://user-images.githubusercontent.com/81375075/128485344-79f6290e-5c73-4dee-b95b-134a661471a7.PNG)
 
So I have a data provider with a table and a button as childrens. The button has an action attached to it that executes a simple `SHOW DATABASES` query but the table doesn't update.

**Comment by mjashanks** [comment]
Hi @VilobatirKedis 

Try putting your Data provider inside your form. I got mine to work, with a structure like this:

![image](https://user-images.githubusercontent.com/3524181/128492611-df38bd76-d85c-461d-ab25-e8276231acef.png)

You should still be able to achieve the same layout as above - just make your form the top-level component.

**Reply by VilobatirKedis** [reply]
Hi,
You mean putting the data provider under the form that I use to create databases? Because I just tried that and it didn't work.

**Reply by mjashanks** [reply]
Hmm - actually, this is a custom SQL query isn't it, for `CREATE DATABASE`. Apologies, I was thinking that you were using the standard table connector.

In that case, I don't think there is anything you can do right now.

I feel like a solution to this would be to provide a "Refresh Datasource" action, which you could trigger after the button click. I will suggest this as a feature.

**Reply by KernOil** [reply]
I also could use that function in the button to refresh the data table from a Mysql database!
